File: 0398-random-pick-index/0398-random-pick-index.py
class Solution:

    def __init__(self, nums: List[int]):
        self.dict = defaultdict(list)
        for index,num in enumerate(nums):
            self.dict[num].append(index)
        print(self.dict)
    def pick(self, target: int) -> int:
        if self.dict[target] and len(self.dict[target]) > 0:
            indices = self.dict[target]
            randomInx = random.choice(indices)
            return randomInx


# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.pick(target)

File: 0398-random-pick-index/0398-random-pick-index.ts
type Dict = {
    [key: number]: number[]; 
}

class Solution {
    private dict: Dict;
    private index: number;

    constructor(nums: number[]) {
        this.dict = {}; 
        this.index = 0;

        for (let i = 0; i < nums.length; i++) { 
            const num = nums[i]; 
            if (!this.dict[num]) { 
                this.dict[num] = []; 
            }
            this.dict[num].push(i);
        }
    }

    pick(target: number): number {
        const indices = this.dict[target];
        const randomIndex = indices[Math.floor(Math.random() * indices.length)]
        return randomIndex;
    }
}

/**
 * Your Solution object will be instantiated and called as such:
 * var obj = new Solution(nums);
 * var param_1 = obj.pick(target);
 */


